GitOut/Features/Git/Diff/GitDiffControl.xaml.cs

```diff
@@ -88,156 +85,62 @@ public bool ShowSpacesAsDots
+                            if (context.SourceId is not null)
+                            {
+                                Stream sourceImage = await context.GetSourceStreamAsync();
+                                control.CurrentContent = new ImageViewModel(context.Result.Blob.Stream, sourceImage);
+                            }
+                            else
+                            {
+                                control.CurrentContent = new ImageViewModel(context.Result.Blob.Stream);
+                            }
                         }
-                        section.Blocks.Add(line);
-                        diffContexts.Add((line, text));
                     }
-                    document.Blocks.Add(section);
+                    else if (context.Result.Text is not null)
+                    {
+                        double pixelsPerDip = VisualTreeHelper.GetDpi(control).PixelsPerDip;
+                        DiffDisplayOptions display = control.ShowSpacesAsDots
+                            ? new DiffDisplayOptions(
+                                pixelsPerDip,
+                                (Brush)Application.Current.Resources["MaterialLightDividers"],
+                                (Brush)Application.Current.Resources["MaterialGray400"],
+                                new ShowSpacesAsDotsTransform()
+                            )
+                            : new DiffDisplayOptions(
+                                pixelsPerDip,
+                                (Brush)Application.Current.Resources["MaterialLightDividers"],
+                                (Brush)Application.Current.Resources["MaterialGray400"]
+                            );
+                        var vm = GitDiffViewModel.ParseDiff(context.Result.Text.Hunks, display);
+                        control.CurrentContent = vm;
+                    }
                 }
-                document.PageWidth = maxWidth + 20;
-                return new GitDiffViewModel(document, lineNumbers, diffContexts);
-
-                double CalculateLineWidth(string text) =>
-                    new FormattedText(
-                    display.Transform.Transform(text),
-                    CultureInfo.InvariantCulture,
-                    FlowDirection.LeftToRight,
-                    new Typeface(FontFamilyName),
-                    12,
-                    Brushes.White,
-                    display.PixelsPerDip
-                ).Width;
-
-                static Paragraph CreateHeaderParagraph(string text, Brush foreground) => new(new Run(text))
-                {
-                    Background = Brushes.Transparent,
-                    Foreground = foreground,
-                    FontSize = 11,
-                    Margin = new Thickness(0)
-                };
             }
         }
 
-        private class DiffLineHighlighter : ILineDecorator
+        private static bool IsImageFile(string extension) => new[]
```

Comment:
Optional, since there are 5 elements :D 
but could this be turned into a hashset instead?

Reply:
Good point, I was planning to break this out and maybe inject a dependency that would provide this list instead. I'll make this local for the time being and let's see where we end up in the end.